pkg/controllers/hub/trafficmanagerprofile/controller.go

```diff
@@ -226,41 +226,117 @@ func (r *Reconciler) handleUpdate(ctx context.Context, profile *fleetnetv1beta1.
+	// Compare monitor config
+	if !equalMonitorConfig(current.Properties.MonitorConfig, desired.Properties.MonitorConfig) {
 		return false
 	}
 
-	if *current.Properties.MonitorConfig.IntervalInSeconds != *desired.Properties.MonitorConfig.IntervalInSeconds ||
-		*current.Properties.MonitorConfig.Path != *desired.Properties.MonitorConfig.Path ||
-		*current.Properties.MonitorConfig.Port != *desired.Properties.MonitorConfig.Port ||
-		*current.Properties.MonitorConfig.Protocol != *desired.Properties.MonitorConfig.Protocol ||
-		*current.Properties.MonitorConfig.TimeoutInSeconds != *desired.Properties.MonitorConfig.TimeoutInSeconds ||
-		*current.Properties.MonitorConfig.ToleratedNumberOfFailures != *desired.Properties.MonitorConfig.ToleratedNumberOfFailures {
+	// Compare custom headers
+	if !equalCustomHeaders(current.Properties.MonitorConfig.CustomHeaders, desired.Properties.MonitorConfig.CustomHeaders) {
 		return false
 	}
 
-	if *current.Properties.ProfileStatus != *desired.Properties.ProfileStatus || *current.Properties.TrafficRoutingMethod != *desired.Properties.TrafficRoutingMethod {
+	// Compare profile status and traffic routing method
+	if !equalProfileProperties(current.Properties, desired.Properties) {
 		return false
 	}
 
-	if current.Properties.DNSConfig.TTL == nil || *current.Properties.DNSConfig.TTL != *desired.Properties.DNSConfig.TTL {
+	// Compare DNS config
+	if !equalDNSConfig(current.Properties.DNSConfig, desired.Properties.DNSConfig) {
 		return false
 	}
 
-	if current.Tags == nil {
+	// Compare tags
+	if !equalTags(current.Tags, desired.Tags) {
+		return false
+	}
+
+	return true
+}
+
+// hasRequiredProperties checks if the profile has all required properties
+func hasRequiredProperties(profile armtrafficmanager.Profile) bool {
+	return profile.Properties != nil &&
+		profile.Properties.MonitorConfig != nil &&
+		profile.Properties.ProfileStatus != nil &&
+		profile.Properties.TrafficRoutingMethod != nil &&
+		profile.Properties.DNSConfig != nil
+}
+
+// equalMonitorConfig compares the monitoring configuration
+func equalMonitorConfig(current, desired *armtrafficmanager.MonitorConfig) bool {
+	if current.IntervalInSeconds == nil || current.Path == nil ||
+		current.Port == nil || current.Protocol == nil ||
+		current.TimeoutInSeconds == nil || current.ToleratedNumberOfFailures == nil {
+		return false
+	}
+
+	return *current.IntervalInSeconds == *desired.IntervalInSeconds &&
+		*current.Path == *desired.Path &&
+		*current.Port == *desired.Port &&
+		*current.Protocol == *desired.Protocol &&
+		*current.TimeoutInSeconds == *desired.TimeoutInSeconds &&
+		*current.ToleratedNumberOfFailures == *desired.ToleratedNumberOfFailures
+}
+
+// equalCustomHeaders compares custom headers between profiles
+func equalCustomHeaders(currHeaders, desiredHeaders []*armtrafficmanager.MonitorConfigCustomHeadersItem) bool {
+	// If lengths are different, headers are not equal
+	if len(currHeaders) != len(desiredHeaders) {
```

Comment:
use the equality.Semantic.DeepEqual in the k8s.io/apimachinery/pkg/api/equality package instead